app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class OffersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_active_subscription!
+
+  def new
+    @offer = Offer.new(conversation: @conversation)
+  end
+
+  def create
+    user_not_authorized unless conversation.business?(current_user)
+
+    @offer = Offer.new(**offer_params, conversation_id: params[:conversation_id])
+    @offer.save_and_notify
+
+    redirect_to conversation_path(params[:conversation_id])
+  end
+
+  def accept
+    user_not_authorized unless conversation.developer?(current_user)
+
+    offer.accept_and_notify
+
+    redirect_to conversation_path(conversation)
+  end
+
+  def decline
+    user_not_authorized unless conversation.developer?(current_user)
+
+    offer.decline_and_notify
+
+    redirect_to conversation_path(conversation)
+  end
+
+  private
+
+  def require_active_subscription!
+    if conversation.business?(current_user) && !current_user.permissions.active_subscription?
+      redirect_to pricing_path, alert: t("errors.business_subscription_inactive")
+    end
+  end
```

Comment:
If you do so, you are violating the ISP principle, "I" from SOLID, which say:
`No class should be forced to implement any interface method(s) that it does not use.`

In that case a better approach is to think on that `require_active_subscription` as a "role", for example: `ActiveSubscriptionRequirer` (Read about Duck Typing in Ruby).

That role is the behavior shared between some specific classes, so just that role must be shared.

A possible solution is to create a concern and include that concern into those controllers that have this behavior in common.

Reply:
I do not agree. ISP principle relates to the interface's methods implementation. If we put `require_active_subscription`  in the parent class we actually implement this method through inheritance. We do not force anything.
There are no interfaces in Ruby (at least not in the language per se).

I do agree using a module is a better approach tho.

Reply:
We use `require_active_subscription` in three different places. Only in two, the implementation is the same. I can see your concern with moving the method into `ApplicationController` now.

It would require a little bit more changes to refactor it than I expected.
Should we leave it, and delegate another issue for checking subscription around the controllers, or should I extract these two duplicate implementations into a concern now?